dspace/modules/payment-system/payment-api/src/main/java/org/dspace/workflow/actions/processingaction/FinalPaymentAction.java

```diff
@@ -79,39 +80,35 @@ public ActionResult execute(Context c, WorkflowItem wfi, Step step, HttpServletR
             }
         }
 
-	// if journal-based subscription is in place, transaction is paid
-	if(shoppingCart.getJournalSub()) {
-	    log.info("processed journal subscription for Item " + itemID + ", journal = " + shoppingCart.getJournal());
-	    log.debug("tally credit for journal = " + shoppingCart.getJournal());
-	    String success = "";
-	    Scheme scheme = Scheme.findByIdentifier(c,ConfigurationManager.getProperty("solrauthority.searchscheme.prism_publicationName"));
-	    Concept[] concepts = Concept.findByPreferredLabel(c,shoppingCart.getJournal(),scheme.getID());
-	    if(concepts!=null&&concepts.length!=0){
-		AuthorityMetadataValue[] metadataValues = concepts[0].getMetadata("journal", "customerID", null, Item.ANY);
-            if (metadataValues!=null&&metadataValues.length>0) {
-                try {
-                    String packageDOI = DOIIdentifierProvider.getDoiValue(wfi.getItem());
-                    shoppingCart.setStatus(ShoppingCart.STATUS_COMPLETED);
-                    Date date= new Date();
-                    shoppingCart.setPaymentDate(date);
-                    shoppingCart.update();
-                    success = AssociationAnywhere.tallyCredit(c, metadataValues[0].value, packageDOI);
-                    paymentSystemService.sendPaymentApprovedEmail(c,wfi,shoppingCart);
-                    return new ActionResult(ActionResult.TYPE.TYPE_OUTCOME, ActionResult.OUTCOME_COMPLETE);
-                } catch (Exception e) {
-                    log.error(e.getMessage(),e);
-                    paymentSystemService.sendPaymentErrorEmail(c, wfi, shoppingCart, "problem: credit not tallied successfully. \\n \\n " + e.getMessage());
-                    return new ActionResult(ActionResult.TYPE.TYPE_OUTCOME, 2);
+            DryadOrganizationConcept organizationConcept = shoppingCart.getSponsoringOrganization(c);
+            // if journal-based subscription is in place, transaction is paid
+            if (organizationConcept != null) {
+                if (shoppingCart.getJournalSub()) {
```

Comment:
This line is a little hard to read. Can we change the name of getJournalSub() to hasSubscription() ?